verifier/src/lib.rs

```diff
@@ -257,16 +255,16 @@ where
 
 impl<T, E> Verifier<E> for EqualityVerifier<T>
 where
-    T: Debug + Clone + PartialEq + IntoVerificationError,
+    T: Debug + Clone + PartialEq,
     E: Accessor<T>,
 {
-    type Error = VerificationError;
-    fn verify(&self, evidence: &E) -> CtOption<Self::Error> {
+    type Value = T;
+    fn verify(&self, evidence: &E) -> VerificationOutput<Self::Value> {
         let expected = self.expected.clone();
         let actual = evidence.get();
         // TODO - This should be a constant time comparison.
-        let is_some = if expected == actual { 0 } else { 1 };
-        CtOption::new(T::into_verification_error(expected, actual), is_some.into())
+        let is_success = if expected == actual { 1 } else { 0 };
+        VerificationOutput::new(actual, is_success.into())
```

Comment:
Note: one benefit of this is it gets away from the negative logic that `CtOption` imposed.
when a `CtOption` was "Some" it meant it held an error or a failed verification.